### PR TITLE
🐛 Ensure Table/Model Data Refresh After Publish

### PR DESCRIPTION
**Pull request by mistryrn**
* Fixes a bug where table/model data would not refresh after very fast publishes

### CODE REVIEW

File: ui/src/components/admin/Model/ModelSingleHeader.js [hunk 1]
```diff
@@ -100,20 +101,19 @@ const ModelSingleHeader = ({ modelName }) => {
   useEffect(() => {
     const refreshModelData = async () => await fetchModelData(modelName);
 
-    if (modelName && isPublishingModel(modelName) && (!publishState || !publishState.current)) {
```
Comment (mistryrn) [inline ui/src/components/admin/Model/ModelSingleHeader.js:103]: Removing the check on `publishState` here since we also need this to set back to `'publishing'` when you re-trigger a publish or publish the same model multiple times without refreshing

File: ui/src/components/admin/Model/ModelSingleHeader.js [hunk 2]
```diff
@@ -100,20 +101,19 @@ const ModelSingleHeader = ({ modelName }) => {
...
+    if (modelName && isPublishingModel(modelName)) {
       publishState.current = 'publishing';
     }
 
     if (
       modelName &&
-      publishState.current === 'publishing' &&
+      (publishState.current === 'publishing' || !publishState.current) &&
```
Comment (mistryrn) [inline ui/src/components/admin/Model/ModelSingleHeader.js:110]: Adding a check for `!publishState.current` since some publishes are completing faster than the time it takes to set the state to `'publishing'` (i.e. by the time this useEffect is called, the publish has already finished. We still need to refresh the model data in this case, otherwise the page is left with a "Changes Not Yet Published" label)

By also allowing `!publishState.current`, we ensure that we refresh the model data even when the publish happens too fast.

File: ui/src/components/admin/Model/ModelSingleHeader.js [hunk 3]
```diff
@@ -100,20 +101,19 @@ const ModelSingleHeader = ({ modelName }) => {
...
       publishProgress.success.find(model => model.modelName === modelName)
     ) {
       publishState.current = 'published';
       refreshModelData();
-      return;
     }
-  }, [publishRunning, publishProgress, modelName, isPublishingModel, fetchModelData]);
```
Comment (mistryrn) [inline ui/src/components/admin/Model/ModelSingleHeader.js:116]: Removing unnecessary deps (added the linter ignore flag atop the file) since they were causing the useEffect to run unnecessarily.

File: ui/src/components/admin/Notifications/PublishProgress.js [hunk 4]
```diff
@@ -162,22 +162,17 @@ const PublishProgress = ({ renderIcon }) => {
     };
 
     useEffect(() => {
-      if (!prevPublishState) {
```
Comment (mistryrn) [inline ui/src/components/admin/Notifications/PublishProgress.js:165]: Similar to the other file, we're removing the null check here

File: ui/src/components/admin/Notifications/PublishProgress.js [hunk 5]
```diff
@@ -162,22 +162,17 @@ const PublishProgress = ({ renderIcon }) => {
...
-        prevPublishState.current = getBulkPublishState();
-        return;
-      }
-
       if (
         refreshModelsTable &&
         (getBulkPublishState() === BulkPublishState.complete ||
           getBulkPublishState() === BulkPublishState.stopped) &&
-        prevPublishState.current === BulkPublishState.publishing
+        (prevPublishState.current === BulkPublishState.publishing || !prevPublishState.current)
```
Comment (mistryrn) [inline ui/src/components/admin/Notifications/PublishProgress.js:169]: Also similarly to the other file, adding a null check here so that we trigger the refresh in the event that the publish finishes before this useEffect is called, ensuring the table data gets updated after it completes.

**Review by mistryrn — COMMENTED** [review]
TL;DR: some publishes are happening so fast that the useEffects aren't able to set the "prevState" to compare against in order to determine if the data needs to be refreshed.

By the time these useEffects are called the publish is already complete, but because they haven't set the "prevState" the conditions aren't met to trigger the refresh.

These updates just move the null checks around and clean up the dependency lists of the useEffects. In my local testing this resolves the issues and doesn't cause any new ones.

**Review by joneubank — APPROVED** [review]
